### PR TITLE
Removal of all requirement text files 

### PR DESCRIPTION
**Pull request by nabobalis**
Has the work from #2810 as the first commit.
Renamed `Maps` to `Map` for example gallery
Redo of the readmes in several folders.
Deleted the requirements folder as it all now lives in setup.cfg
Deleted the readme and make files in the docs folder. 
Added `docs` to the setup.py. 
RTD will now use this to build the project instead of any yaml file. 
The long description is now the readme file and will be displayed in PyPi. [See here](https://test.pypi.org/project/sunpy/1.0.0.dev9991/) 
Robots should also only find the best doc versions now.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ docs-install: &docs-install
   command: |
     pip install -r requirements/docs.txt
```
Comment (nabobalis) [inline .circleci/config.yml]: DOH.

File: docs/conf.py [hunk 2]
```diff
@@ -43,6 +43,14 @@
     warninghandler = list(filter(lambda x: isinstance(x, WarningStreamHandler), handlers))[0]
     warninghandler.stream = SafeEncodingWriter(stream=sys.stdout)
 
+# -- I hear you like changelogs in your conf files -----------------------------
```
Comment (Cadair) [inline docs/conf.py]: I would just jam this all in one block at the end, no need to put it in the top and the bottom.

**Comment by Cadair** [comment]
lmao, I thought the time PR fixed that test?

**Comment by nabobalis** [comment]
No it still fails maybe 1 in 100 times.

**Comment by Cadair** [comment]
Well we haven't merged the time PR yet... ?

**Comment by nabobalis** [comment]
No? I'm confused what you are referring to now.

**Pull request merged**